### ISSUE TITLE
Bug in 4.0 (beta)

### ISSUE THREAD

**Issue by xlovinglyx** [lieser/X-Unsent_support#9]
if i reopen thunderbird, the x-unsent function is lost, until i reinstall the addon, each time i reopen tb again .. .

**Comment by lieser** [comment]
I just restated Thunderbird multiple times and fail to reproduce this.

Can you please take a look if the error console reports any errors?

**Comment by xlovinglyx** [comment]
Error: Cannot find loader for bootstrap XPIProvider.jsm:1887:15
Error while loading 'jar:file:///C:/Program%20Files/Mozilla%20Thunderbird/omni.ja!/chrome/messenger/search-extensions/twitter/manifest.json' (NS_ERROR_FILE_NOT_FOUND) Extension.jsm:570
    readJSON resource://gre/modules/Extension.jsm:570
    onStopRequest resource://gre/modules/NetUtil.jsm:128

[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
    loadSync resource://gre/modules/L10nRegistry.jsm:658
    fetchFile resource://gre/modules/L10nRegistry.jsm:573
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:478
    map self-hosted:240
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:473
    generateResourceSetsForLocaleSync resource://gre/modules/L10nRegistry.jsm:415
    InterpretGeneratorResume self-hosted:1151
    next self-hosted:1099
    generateBundlesSync resource://gre/modules/L10nRegistry.jsm:177
    InterpretGeneratorResume self-hosted:1151
    next self-hosted:1099
    touchNext resource://gre/modules/Localization.jsm:167
    regenerateBundles resource://gre/modules/Localization.jsm:552
    activate resource://gre/modules/Localization.jsm:243
    <anonymous> chrome://messenger/content/mailWidgets.js:37
    <anonymous> chrome://messenger/content/customElements.js:34
    <anonymous> chrome://messenger/content/customElements.js:37
    observe resource://gre/modules/MailGlue.jsm:201

Components.manager.registerFactory is not a function commandLineHandler.js:354
    init jar:file:///C:/Users/Administrator/AppData/Roaming/Thunderbird/Profiles/su7pmwuk.default/extensions/X-Unsent_support@pl.xpi!/components/commandLineHandler.js:354
    init self-hosted:844
    result resource://gre/modules/ExtensionParent.jsm:996
    withPendingBrowser resource://gre/modules/ExtensionParent.jsm:602
    result resource://gre/modules/ExtensionParent.jsm:996
    callAndLog resource://gre/modules/ExtensionParent.jsm:958
    recvAPICall resource://gre/modules/ExtensionParent.jsm:995
    AsyncFunctionNext self-hosted:693

[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
    loadSync resource://gre/modules/L10nRegistry.jsm:658
    fetchFile resource://gre/modules/L10nRegistry.jsm:573
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:478
    map self-hosted:240
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:473
    generateResourceSetsForLocaleSync resource://gre/modules/L10nRegistry.jsm:415
    next self-hosted:1099
    generateBundlesSync resource://gre/modules/L10nRegistry.jsm:177
    next self-hosted:1099
    touchNext resource://gre/modules/Localization.jsm:167
    regenerateBundles resource://gre/modules/Localization.jsm:552
    activate resource://gre/modules/Localization.jsm:243
    <anonymous> chrome://openpgp/content/modules/trust.jsm:11
    <anonymous> chrome://openpgp/content/modules/keyRing.jsm:17
    <anonymous> chrome://openpgp/content/modules/windows.jsm:19
    loader chrome://openpgp/content/modules/lazy.jsm:20
    getService chrome://openpgp/content/modules/core.jsm:426
    getService chrome://openpgp/content/modules/core.jsm:183
    init chrome://openpgp/content/BondOpenPGP.jsm:100
    InterpretGeneratorResume self-hosted:1151
    AsyncFunctionNext self-hosted:693

[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
    loadSync resource://gre/modules/L10nRegistry.jsm:658
    fetchFile resource://gre/modules/L10nRegistry.jsm:573
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:478
    map self-hosted:240
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:473
    generateResourceSetsForLocaleSync resource://gre/modules/L10nRegistry.jsm:415
    next self-hosted:1099
    generateBundlesSync resource://gre/modules/L10nRegistry.jsm:177
    next self-hosted:1099
    touchNext resource://gre/modules/Localization.jsm:167
    regenerateBundles resource://gre/modules/Localization.jsm:552
    activate resource://gre/modules/Localization.jsm:243
    <anonymous> resource:///modules/calendar/calCalendarDeactivator.jsm:10
    <anonymous> chrome://calendar/content/calendar-chrome-startup.js:27

[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
    loadSync resource://gre/modules/L10nRegistry.jsm:658
    fetchFile resource://gre/modules/L10nRegistry.jsm:573
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:478
    map self-hosted:240
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:473
    generateResourceSetsForLocaleSync resource://gre/modules/L10nRegistry.jsm:415
    next self-hosted:1099
    generateBundlesSync resource://gre/modules/L10nRegistry.jsm:177
    next self-hosted:1099
    touchNext resource://gre/modules/Localization.jsm:167
    regenerateBundles resource://gre/modules/Localization.jsm:552
    activate resource://gre/modules/Localization.jsm:243
    <anonymous> resource:///modules/OTR.jsm:15
    <anonymous> resource:///modules/OTRUI.jsm:8
    connectedCallback chrome://messenger/content/chat/chat-conversation-info.js:117
    <anonymous> chrome://global/content/customElements.js:217

[Exception... "Component returned failure code: 0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH) [nsIXPCComponents_Utils.readUTF8URI]"  nsresult: "0x80520001 (NS_ERROR_FILE_UNRECOGNIZED_PATH)"  location: "JS frame :: resource://gre/modules/L10nRegistry.jsm :: L10nRegistry.loadSync :: line 658"  data: no] L10nRegistry.jsm:658:19
    loadSync resource://gre/modules/L10nRegistry.jsm:658
    fetchFile resource://gre/modules/L10nRegistry.jsm:573
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:478
    map self-hosted:240
    generateResourceSetSync resource://gre/modules/L10nRegistry.jsm:473
    generateResourceSetsForLocaleSync resource://gre/modules/L10nRegistry.jsm:415
    next self-hosted:1099
    generateBundlesSync resource://gre/modules/L10nRegistry.jsm:177
    next self-hosted:1099
    touchNext resource://gre/modules/Localization.jsm:167
    regenerateBundles resource://gre/modules/Localization.jsm:552
    activate resource://gre/modules/Localization.jsm:243
    <anonymous> resource:///modules/OTRUI.jsm:10
    connectedCallback chrome://messenger/content/chat/chat-conversation-info.js:117
    <anonymous> chrome://global/content/customElements.js:217

**Comment by xlovinglyx** [comment]
Tb v78.5.0

**Comment by lieser** [comment]
Thanks for the output. The relevant part is:
```
Components.manager.registerFactory is not a function commandLineHandler.js:354
```
Seems like you are running in a timing issue. Will try make a fix in the next few days for you to try out.

**Comment by xlovinglyx** [comment]
Ok then, 

& if all other addons are disabled, or only Get/Send Button, is added, or i reinstall x-unsent, x-unsent works.

All others i have:

Auto Select Latest Message
Compact Headers
Deselect on Delete TB78
LookOut (fix version)
Mail Merge
SubSwitch

**Comment by lieser** [comment]
Can still not reproduce it myself, also with the same extensions installed. But I should have hopefully still fixed it. Can you please try the attached version, to confirm that it works for you?

---
edit: removed attachment

**Comment by lieser** [comment]
Sorry, the version I attached above contained also a second workaround, which should not be needed.

Can you please do you test with the one I attached here? And please let me know if you tested it successfully. Thanks a lot.
[X-Unsent_support@pl.zip](https://github.com/lieser/X-Unsent_support/files/5610494/X-Unsent_support%40pl.zip)

**Comment by xlovinglyx** [comment]
yep, this issue's fixed.

**Comment by lieser** [comment]
Thanks for the quick feedback. Just released version 4.0.1 with the fix included.
